library/std/src/Std/Diagnostics.qs

```diff
@@ -122,7 +122,6 @@ function DumpMatrix(qs : Qubit[]) : Unit {
 /// # Remarks
 /// This operation is useful for checking whether a qubit is in the |0⟩ state during simulation. It is not possible to check
 /// this on hardware without measuring the qubit, which could change the state.
-@Config(Unrestricted)
 operation CheckZero(qubit : Qubit) : Bool {
```

Comment:
I'm not sure I understand. Suppose I use Base profile. If I call this from a regular operation where will it fail compilation? And if I call it from a @Test() operation it will run correctly and perform the check? Can @Test function be called from a regular one?

Reply:
OK, I cannot call `@Test` operations from regular ones, good.
What about CheckZero? Can I call it now in base profile from a normal operation. Before this change it wouldn't be available.

Reply:
No, it will generate an error in Base profile:
![image](https://github.com/user-attachments/assets/347568b1-4081-402b-a521-aa054ec32c1e)
For Adaptive, it won't be a compile time error but an error during QIR code generation (see the image shared with Bill here: https://github.com/microsoft/qsharp/pull/2584#discussion_r2195671191)